### PR TITLE
Clean up catalogName a bit

### PR DESCRIPTION
**Pull request by paveltomin**
catalog name related piece from https://github.com/GEOS-DEV/GEOS/pull/3021/

### CODE REVIEW

**Comment by rrsettgast** [comment]
@paveltomin Did you search the whole codebase? There may be more places

File: src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoromechanics.hpp [hunk 1]
```diff
@@ -52,7 +53,21 @@ class SinglePhasePoromechanics : public PoromechanicsSolver< FLOW_SOLVER, MECHAN
...
+  {
+    return "SinglePhasePoromechanics";
+  }
+  template< typename _FLOW_SOLVER=FLOW_SOLVER >
+  static
+  typename std::enable_if< std::is_same< _FLOW_SOLVER, SinglePhaseReservoirAndWells< SinglePhaseBase > >::value, string >::type
+  catalogName()
+  {
+    return SinglePhaseReservoirAndWells< SinglePhaseBase >::catalogName() + "Poromechanics";
+  }
```
Comment (TotoGaz) [inline src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoromechanics.hpp]:
```suggestion
 static string catalogName()
  {
      if constexpr ( std::is_same_v< FLOW_SOLVER, SinglePhaseBase > )
      {
          return "SinglePhasePoromechanics";
      }
      else if constexpr( std::is_same_v< FLOW_SOLVER, SinglePhaseReservoirAndWells< SinglePhaseBase > > )
      {
          return SinglePhaseReservoirAndWells< SinglePhaseBase >::catalogName() + "Poromechanics";
      }
      else
      {
          static_assert(false, "blah blah");
      }
  }
```
Is this working?

Comment (paveltomin) [reply src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoromechanics.hpp]: i took it from here to be honest: https://github.com/GEOS-DEV/GEOS/blob/ddf3e97a398c1261cae81cf2f864112061e5b259/src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseFVM.hpp#L101
can you your version if it's preferred

Comment (TotoGaz) [reply src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoromechanics.hpp]: If my version is working and doing the right thing, I think it's more modern and simple.
I think that the `if constexpr` c++17 feature was thought for this use case (at least).

Comment (paveltomin) [reply src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoromechanics.hpp]: that `static_assert` fails for some reason, not sure why

Comment (TotoGaz) [reply src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoromechanics.hpp]: My `static_assert` code was bad, I wrote it very fast and surely you need to be something a little smarter!

Comment (paveltomin) [reply src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoromechanics.hpp]: i put together something, let's see if it can pass the tests

**Comment by paveltomin** [comment]
> @paveltomin Did you search the whole codebase? There may be more places

i looked through yes but would not guarantee i paid full attention
mostly looked into solvers part

File: src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp [hunk 2]
```diff
@@ -35,38 +35,17 @@ using namespace constitutive;
...
 {
-  return HydrofractureSolverCatalogNames< POROMECHANICS_SOLVER >::name();
+  return "Hydrofracture";
 }
-
+//template<>
+//string HydrofractureSolver< MultiphasePoromechanics< CompositionalMultiphaseBase > >::catalogName()
+//{
+//  return "MultiphaseHydrofracture";
+//}
```
Comment (TotoGaz) [inline src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp]: You do not want to spread the pattern for this function as well?

File: src/coreComponents/physicsSolvers/multiphysics/SinglePhaseReservoirAndWells.hpp [hunk 3]
```diff
@@ -26,13 +26,13 @@
 namespace geos
 {
 
-template< typename SINGLEPHASE_RESERVOIR_SOLVER >
-class SinglePhaseReservoirAndWells : public CoupledReservoirAndWellsBase< SINGLEPHASE_RESERVOIR_SOLVER,
+template< typename RESERVOIR_SOLVER >
```
Comment (TotoGaz) [inline src/coreComponents/physicsSolvers/multiphysics/SinglePhaseReservoirAndWells.hpp:30]: Please add a `@tparam` that emphasize that we need a single phase solver (or the constraint is not valid anymore)?

**Review by TotoGaz — APPROVED** [review]

File: src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp (hunk 2 above)
Comment (paveltomin) [reply src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp]: just did, thanks

File: src/coreComponents/physicsSolvers/multiphysics/SinglePhaseReservoirAndWells.hpp (hunk 3 above)
Comment (paveltomin) [reply src/coreComponents/physicsSolvers/multiphysics/SinglePhaseReservoirAndWells.hpp:30]: added, it still has to be single phase yes

**Review by rrsettgast — APPROVED** [review]

**Pull request merged**